.circleci/config.yml

```diff
@@ -133,7 +156,104 @@ jobs:
           root: ~/
           paths:
             - project
+            - dist/*.kar
+
+  smoke-test:
+    executor: smoke-test-executor
+
+    steps:
+      - attach_workspace:
+          at: ~/
   
+      - restore_cache:
+          keys:
+          # attempt to use the cache from the last smoke-test run
+          - v2-smoke-dependencies-{{ checksum "smoke-test/pom.xml" }}-{{ checksum "pom.xml" }}
+          # use the cache built for the same root pom
+          - v4-dependencies-{{ checksum "pom.xml" }}
+          # fallback to using the latest cache if no exact match is found
+          - v4-dependencies-
+          
+      - restore_cache:
+          keys:
+          - v1-docker-{{ checksum "smoke-test/src/main/resources/docker_fixed_images" }}
+  
+      # Link the cached maven repo
+      # Note: this is needed since we are using a circleci vm that is running as circleci whereas our images run as root
+      - run:
+          name: Link maven repo with cache
+          command: |
+            ln -s /tmp/.m2 ~/.m2
+  
+      - run:
+          name: Make the kar available
+          command: |
+            mkdir -p ~/assembly/kar/target/
+            ln -s ~/dist/opennms-oce-plugin.kar ~/assembly/kar/target/opennms-oce-plugin.kar
+
+      - run:
```

Comment:
Just some food for thoughts here using floating tags. I would suggest avoiding floating tags on external dependencies which are not built artifacts from this repository, e.g. Horizon, Minion, Sentinel. We should fix them to specific versioned build artifacts. Otherwise, we have additional uncontrolled moving parts interfere with the tests.

For example, you make a code change in OCE and the floating tag moved on in the Horizon image. The test fails and now you have to diagnose two parts and if the Horizon tag moves on again you can't reproduce it anymore and it magically got fixed or got broken differently.

Reply:
How often will fixed version containers be published? I used the floating ones for now because those had the latest changes we wanted to test with. Once there are fixed containers containing what we need, we can switch to them.

I think ultimately we probably want to test with both sets. Perhaps the on-commit job can used a fixed version and a nightly test-run can use the latest floating version. That way we can catch issues earlier as I'm not aware of any other test coverage testing the distributed install end to end.

Reply:
Agreed. We can switch to fixed tags once we tag and release 24.0.0.

We can then setup additional jobs, perhaps nightly?, that test against floating tags of 25.0.0.

Reply:
> How often will fixed version containers be published?

Can be more often than the release of the software product itself, just as an example the changes Alejandro introduced to run as non-root and when we change other surroundings like behavior with configuration etc.